### PR TITLE
Import Arithmetic library

### PR DESCRIPTION
**Pull request by ssomayyajula**
### Description

Completed:
- Moves the NonlinearArithmetic `library` from the libraries repo into `Source/DafnyStandardLibraries`
- Removes `RUN` and `include` directives except for those in `Arithmetic/Internals/DivInternals.dfy` and `Mul.dfy` referencing `Arithmetic/Internals/*Nonlinear.dfy` (see those files for more details)
- Fix: proof of `LemmaPowPositive` in `Power.dfy`
- Style: Replaces `forall ensures` with `assert by` everywhere 
- Adds Robin’s build process for this library, edits it to account for `Internals/*Nonlinear.dfy` needing non-linear arithmetic (see the `.toml` files for more details)
- `LemmaDivAuto()` in `DivMod.dfy` brought under resource limit
- Standards: triggers added in `DivMod.dfy`

Holding off:
- Standards: `warn-redundant-assumptions` and `warn-contradictory-assumptions` actually breaks a one/two proofs each in `DivMod.dfy`, `Power.dfy`, and `ModInternals.dfy` (the latter breaks a proof in an straightforward-to-fix way, but the former may be flagging false positives with the way proofs are currently written)
- Style: expand short names everywhere

### How has this been tested?

Adding to `examples/ArithmeticExamples.dfy`.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyDriver/Commands/DafnyCli.cs
Comment (robin-aws) [inline Source/DafnyDriver/Commands/DafnyCli.cs:35]: I know this was from my own commit :) but I think we should use a more accurate suffix here, since this doo file only contains part of the Arithmetic library. Perhaps `-nla-disabled.doo`?

File: Source/DafnyStandardLibraries/src/DafnyStdLibs/Arithmetic/README.md
Comment (robin-aws) [inline Source/DafnyStandardLibraries/src/DafnyStdLibs/Arithmetic/README.md:4]:
```suggestion
While wonderful when they work, these heuristics can lead to brittle proofs.  Hence, in many projects,
```

File: Source/DafnyStandardLibraries/src/dfyconfig-arithmetic.toml
Comment (robin-aws) [inline Source/DafnyStandardLibraries/src/dfyconfig-arithmetic.toml:27]: I would keep this one on even if it's reasonable to remove the assumptions ones, since otherwise end users may see other warnings when they flip on `--standard-libraries`

File: Source/DafnyStandardLibraries/src/dfyconfig.toml
Comment (robin-aws) [inline Source/DafnyStandardLibraries/src/dfyconfig.toml:3]:
```suggestion
excludes = ["**/Arithmetic/*.dfy", "**/Arithmetic/**/*Internals.dfy"]
```

Just to keep the two project files 100% in sync. Not THAT critical because if we double-include or miss files, we'll get errors when testing the examples, but better to avoid the possibility. :)

File: Source/DafnyStandardLibraries/src/DafnyStdLibs/Arithmetic/DivMod.dfy
Comment (ssomayyajula) [reply Source/DafnyStandardLibraries/src/DafnyStdLibs/Arithmetic/DivMod.dfy:13]: Nitpicky question: do we prefer verbs or nouns, i.e., `DividesModulo` vs. `DivisionModulus`?

File: Source/DafnyDriver/Commands/DafnyCli.cs
Comment (ssomayyajula) [reply Source/DafnyDriver/Commands/DafnyCli.cs:35]: Agreed; would hold off until PR is merged in the interest of time if that's OK with you.

Comment (robin-aws) [reply Source/DafnyDriver/Commands/DafnyCli.cs:35]: Oh that's just fine, since this isn't user-facing

File: Source/DafnyStandardLibraries/src/DafnyStdLibs/Arithmetic/DivMod.dfy
Comment (robin-aws) [reply Source/DafnyStandardLibraries/src/DafnyStdLibs/Arithmetic/DivMod.dfy]: I think that's fine, the nesting is the much more critical point anyway.

File: Source/DafnyStandardLibraries/examples/dfyconfig.toml
Comment (ssomayyajula) [reply Source/DafnyStandardLibraries/examples/dfyconfig.toml:7]: Should be back in the latest version!

File: Source/DafnyStandardLibraries/src/dfyconfig-arithmetic.toml
Comment (ssomayyajula) [reply Source/DafnyStandardLibraries/src/dfyconfig-arithmetic.toml]: Should be back in the latest version!

File: Source/DafnyStandardLibraries/examples/dfyconfig.toml
Comment (robin-aws) [reply Source/DafnyStandardLibraries/examples/dfyconfig.toml:7]: Looks like it's still there? :)

**Review by robin-aws — APPROVED** [review]
Awesome work, this was one of the tougher ones!

**Pull request merged**
